### PR TITLE
Fix issue #9296 (calling register on KeysView subclass causes false positive error)

### PR DESCRIPTION
**Pull request by jab**
As per discussion in #9296, this PR reverts the commits in #8977 and #9058 that resulted in false positive errors when e.g. calling register() on a KeysView subclass, until a non-buggy implementation is landed. This also adds a regression test.

### CODE REVIEW

File: stdlib/builtins.pyi [hunk 1]
```diff
@@ -804,7 +804,7 @@ class bytearray(MutableSequence[int], ByteString):
     def __alloc__(self) -> int: ...
 
 @final
-class memoryview(Sequence[int]):
+class memoryview(Sized, Sequence[int]):
```
Comment (JelleZijlstra) [inline stdlib/builtins.pyi:807]: Is this change necessary? I would prefer removing explicit protocol base classes if possible, since memoryview doesn't inherit from Sized at runtime.

Comment (jab) [reply stdlib/builtins.pyi:807]: I'm not sure, I was following would I thought was the consensus after I saw https://github.com/python/typeshed/issues/9296#issuecomment-1332377034 (which suggested including this reversion), and then 10 days having gone by without a differing suggestion.

Can you tell me how if you want me to try undoing this and verifying that the regression is still fixed locally before I push again? I tried adding some garbage to the regression test here locally and then re-running `python tests/pyright_test.py` but it still passed, so I'm not sure how to check this.

Comment (JelleZijlstra) [reply stdlib/builtins.pyi:807]: Let's just undo this change and see if CI succeeds.

```suggestion
class memoryview(Sequence[int]):
```

**Review by JelleZijlstra — APPROVED** [review]

File: stdlib/builtins.pyi (hunk 1 above)
Comment (jab) [reply stdlib/builtins.pyi]: Thanks, @JelleZijlstra. Btw, looks like the message of the commit that you merged to main still includes `Revert "typing: remove metaclass from Sized (#9058). This reverts commit a3ce51209544d64d03018b3927051375f1bebe82.`

More importantly, I would still like to know how to check if the regression test I wrote is actually doing its job: Did the tests that passed here in CI actually exercise the code path where this bug was observed? I'm concerned that when I changed "MyKeysView.register" to "MyKeysView.garbage" and tried to have pyright test that code, I didn't see any failures.
